src/System.Private.TypeLoader/src/System/Reflection/Runtime/General/QHandles.cs

```diff
@@ -96,71 +96,51 @@ public override int GetHashCode()
         private readonly Handle _handle;
     }
 
-
-    public struct QTypeDefinition : IEquatable<QTypeDefinition>
+    public partial struct QMethodDefinition
     {
-        public QTypeDefinition(MetadataReader reader, TypeDefinitionHandle handle)
+        private QMethodDefinition(object reader, int token)
         {
             _reader = reader;
-            _handle = handle;
+            _handle = token;
         }
 
-        public MetadataReader Reader { get { return _reader; } }
-        public TypeDefinitionHandle Handle { get { return _handle; } }
-
-        public override bool Equals(Object obj)
+        public static QMethodDefinition FromObjectAndInt(object reader, int token)
         {
-            if (!(obj is QTypeDefinition))
-                return false;
-            return Equals((QTypeDefinition)obj);
+            return new QMethodDefinition(reader, token);
         }
 
-        public bool Equals(QTypeDefinition other)
-        {
-            if (!(_reader == other._reader))
-                return false;
-            if (!(_handle.Equals(other._handle)))
-                return false;
-            return true;
-        }
+        public object Reader { get { return _reader; } }
+        public int Token { get { return _handle; } }
 
-        public override int GetHashCode()
-        {
-            return _handle.GetHashCode();
-        }
+        public bool IsNull { get { return _reader == null; } }
```

Comment:
Based on an offline conversation I've agreed to change this to IsValid instead. That saves IsNull for asking if the token is 0 with a meaningful Reader. (Which is a question we may wish to ask in the future)